exporters/otlp/internal/otlpconfig/options.go

```diff
@@ -69,6 +71,24 @@ const (
 }`
 )
 
+// Keep it in sync with golang's DefaultTransport from net/http! We
+// have our own copy to avoid handling a situation where the
+// DefaultTransport is overwritten with some different implementation
+// of http.RoundTripper or it's modified by other package.
```

Comment:
Is it necessarily problematic if the `DefaultTransport` has been modified elsewhere?  I'm not sure I'm keen on the idea of maintaining our own default configuration that is supposed to stay in sync with the stdlib.

Reply:
This code has been moved from the otlphttp/driver.go:
https://github.com/open-telemetry/opentelemetry-go/pull/1859/files#diff-25cd91630c3b0471c12667c1138ba7715cd6269e539dde179b3b420fe455e2adL52

And I suggest to keep it in to ensure backwards compatability.

Reply:
Yeah, that got added in #1420 with https://github.com/open-telemetry/opentelemetry-go/pull/1420/commits/2cdd01c7ba121646b6add6eb5eda6128706f3c92 based on [a comment](https://github.com/open-telemetry/opentelemetry-go/pull/1420#discussion_r550948799) I made about a check that the `http.DefaultTransport` was still a `*http.Transport` that would `panic()` if it wasn't.  I didn't then, and I'm not sure I do now, understand why it would be problematic if the `http.DefaultTransport` had been changed.  Wouldn't that be indicative that the user really wanted all HTTP clients to use their new default?

I can see it being potentially problematic if the user has replaced `http.DefaultTransport` with `otelhttp.Transport` which will create spans for every outgoing request.  Tracing the requests to ship your tracing data will result in never-ending trace data.  But, that's a misconfiguration the user could create by using the mechanisms provided in this PR as well.  Other SIGs have proposed [a suppression mechanism](https://open-telemetry.github.io/opentelemetry-js/modules.html#suppressinstrumentation) that could be used to indicate that no telemetry should be generated from that context, which seems to me a potentially better solution to that problem.

Reply:
That makes sense. So how can I move forward with this? Should I remove the `DefaultTransport` and fall back to the one from stdlib? And should I implement the suppression mechanism, to prevent the never ending trace scenario? Or would that be out of scope for this PR?

Reply:
Adding a suppression mechanism is definitely out of scope for this PR.  It was discussed briefly at the recent spec SIG meeting but there's not yet consensus on how to handle it.  I think it is something we should be able to add post-1.0 if it becomes required.

For this PR I think we can remove the `DefaultTransport` and simply use the one from stdlib.  @MrAlias thoughts?